main/src/library/Fixpoint/Interpreter.flix

```diff
@@ -102,16 +101,20 @@ namespace Fixpoint {
                 let rel = MutMap.getOrElsePut!(ramSym, new MutMap(Static), db);
                 match toDenotation(ramSym) {
                     case Denotation.Relational =>
-                        let tuple = Tuple(Array.map(evalTerm(env), terms));
+                        let tuple = Tuple(List.toArray(List.map(evalTerm(env), terms), Static));
                         MutMap.put!(tuple, $DEFAULT$, rel)
                     case Denotation.Latticenal(bot, leq, lub, _) =>
-                        let len = Array.length(terms);
-                        let key = Tuple(Array.init(i -> evalTerm(env, terms[i]), len - 1, Static));
-                        let lat = evalTerm(env, terms[len - 1]);
-                        if (lat `leq` bot)
-                            ()
-                        else
-                            MutMap.putWith!(lub, key, lat, rel)
+                        // assume that length(terms) > 0
+                        let len = List.length(terms);
+                        let keyList = terms |> List.map(evalTerm(env));
+                        let (relKeys, latValList) = List.splitAt(len-1, keyList);
+                        let key = Tuple(List.toArray(relKeys, Static));
+                        let latVal = match List.head(latValList) {
```

Comment:
Could we use a non-empty list? `Nel`?

Reply:
Possibly yes. It would be a bit cumbersome though since the required element is the last item of the list so we could perhaps store the terms in reverse. Should I do this?

Reply:
Does Nel not expose a `last` and `init` which is what you want?

Reply:
I would do it now, or at least try, and see how it goes.

Reply:
Ah yes, thats true. Ill give it a go

Reply:
yes id believe so too. ill look it over again